src/Host/Client/Impl/Session/RSessionProvider.cs

```diff
@@ -81,7 +82,14 @@ public class RSessionProvider : IRSessionProvider {
                 return;
             }
 
-            foreach (var session in _sessions.Values) {
+            var sessions = GetSessions();
+            var stopHostTasks = sessions.Select(session => session.StopHostAsync());
+            try {
+                Task.WhenAll(stopHostTasks).GetAwaiter().GetResult();
```

Comment:
If I understand correctly, the chain of `Dispose` calls that gets us here starts with the disposal of some MEF component - won't that necessarily be waited on by MEF?